plugins/outputs/stomp/stomp.go

```diff
@@ -0,0 +1,120 @@
+package stomp
+
+import (
+	ts "crypto/tls"
+	"net"
+
+	"github.com/go-stomp/stomp"
+
+	"github.com/influxdata/telegraf"
+	commontls "github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+)
+
+//STOMP ...
+type STOMP struct {
+	Host      string `toml:"host"`
+	Username  string `toml:"username,omitempty"`
+	Password  string `toml:"password,omitempty"`
+	QueueName string `toml:"queueName"`
+	commontls.ClientConfig
+	conn      *ts.Conn
+	netConn   net.Conn
+	stomp     *stomp.Conn
+	serialize serializers.Serializer
+	Log       telegraf.Logger `toml:"-"`
+}
+
+//Connect ...
+func (q *STOMP) Connect() error {
+	tlsConfig, err := q.ClientConfig.TLSConfig()
+
+	if tlsConfig != nil {
+		q.conn, err = ts.Dial("tcp", q.Host, tlsConfig)
+		if err != nil {
+			return err
+		}
+		q.stomp, err = stomp.Connect(q.conn, stomp.ConnOpt.HeartBeat(0, 0), stomp.ConnOpt.Login(q.Username, q.Password))
+		if err != nil {
+			return err
+		}
+	} else {
+		var err error
+		q.netConn, err = net.Dial("tcp", q.Host)
+		if err != nil {
+			return err
+		}
+		q.stomp, err = stomp.Connect(q.netConn, stomp.ConnOpt.HeartBeat(0, 0), stomp.ConnOpt.Login(q.Username, q.Password))
+		if err != nil {
+			return err
+		}
+
+	}
```

Comment:
This can be simplified as `tls.Dial` returns a struct compatible with the `net.Conn` interface. So we can do
```suggestion
	var err error
	if tlsConfig != nil {
		q.conn, err = ts.Dial("tcp", q.Host, tlsConfig)
		if err != nil {
			return fmt.Errorf("connecting with TLS failed: %s", err)
		}
	} else {
		q.conn, err = net.Dial("tcp", q.Host)
		if err != nil {
			return fmt.Errorf("connecting without TLS failed: %s", err)
		}
	}
	q.stomp, err = stomp.Connect(q.conn, stomp.ConnOpt.HeartBeat(0, 0), stomp.ConnOpt.Login(q.Username, q.Password))
	if err != nil {
		return fmt.Errorf("connecting to STOMP failed: %s", err)
	}
```